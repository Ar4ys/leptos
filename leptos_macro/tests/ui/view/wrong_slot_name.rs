use leptos::*;

#[slot]
struct Then {}

#[slot]
struct ElseIf {}

#[component]
fn If(then: Then) -> impl IntoView {
    _ = then;
}

#[component]
fn SlotIf(then: Then) -> impl IntoView {
    _ = then;
}

fn main() {
    view! {
        <If>
            <ElseIf slot:then />
        </If>
    };

    view! {
        <SlotIf>
            <ElseIf slot:then />
        </SlotIf>
    };
}
