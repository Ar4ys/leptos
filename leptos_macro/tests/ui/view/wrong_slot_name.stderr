error[E0277]: the trait bound `Then: From<ElseIf>` is not satisfied
  --> tests/ui/view/wrong_slot_name.rs:22:13
   |
22 |             <ElseIf slot:then />
   |             ^^^^^^^^^^^^^^^^^^^^ the trait `From<ElseIf>` is not implemented for `Then`
   |
   = note: required for `ElseIf` to implement `Into<Then>`

error[E0277]: the trait bound `Then: From<ElseIf>` is not satisfied
  --> tests/ui/view/wrong_slot_name.rs:28:13
   |
28 |             <ElseIf slot:then />
   |             ^^^^^^^^^^^^^^^^^^^^ the trait `From<ElseIf>` is not implemented for `Then`
   |
   = note: required for `ElseIf` to implement `Into<Then>`
